pkg/gardenlet/controller/federatedseed/namespace/endpoints_controller.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespace
+
+import (
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (c *Controller) endpointAdd(_ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) endpointUpdate(_, _ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) endpointDelete(_ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) enqueueNamespaces() {
+	gardenNamespaceSelector := labels.SelectorFromSet(labels.Set{
+		v1beta1constants.LabelRole: v1beta1constants.GardenNamespace})
+	gardenNamespace, err := c.namespaceLister.List(gardenNamespaceSelector)
+	if err != nil {
+		c.log.Errorf("Failed to update Network Policy %q - unable to get the garden namespace from the lister: %v", ALLOW_TO_SEED_APISERVER, err)
+		return
+	}
+	// expecting only one namespace that is labeled as Gardener namespace
+	if len(gardenNamespace) != 1 {
```

Comment:
Why do you need this check?

Reply:
It is explained one line below in the comment :) 

> 		// happens sometimes on Gardenlet startup when the informer is not synced
		// no problem, as the namespaces (including the garden ns) are enqueued individually

I observed that sometimes during startup, even though it waits until the endpoints & namespace informers are synced, it fails to list the garden namespace by label. It only happens sporadically but is no problem because the NetworkPolicies are synced anyways.